src/components/auth/login-form.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import z from "zod";
+import { Form, FormField, FormMessage } from "../ui/form";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Input } from "../ui/input";
+import { Button } from "../ui/button";
+import { useState, useTransition } from "react";
+import { Label } from "../ui/label";
+import { Mail, Lock } from "lucide-react";
+
+export const loginFormSchema = z.object({
+  email: z.string().email("Ungültige E-Mail-Adresse"),
+  password: z.string().min(6, "Passwort muss mindestens 6 Zeichen lang sein"),
+});
+
+type LoginFormResult = { error?: string } | void;
+
+export function LoginForm({
+  onSubmit,
+}: {
+  onSubmit: (
+    data: z.infer<typeof loginFormSchema>,
+  ) => LoginFormResult | Promise<LoginFormResult>;
```

Comment:
```suggestion
  ) => Promise<LoginFormResult>;
```

Nur ein return type bitte